upgrade/steps-helpers_test.go

```diff
@@ -0,0 +1,48 @@
+package upgrade
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetRepoExpectedLocation(t *testing.T) {
+	ctx := Context{
+		GoPath: "Users/myuser/go",
+	}
+
+	cwd, err := os.Getwd()
+	assert.Nil(t, err)
+
+	// test cwd == repo path
+	mockRepoPath := cwd
+	expected, err := getRepoExpectedLocation(ctx, mockRepoPath)
+	mockRepoPath = trimSeparators(mockRepoPath)
+	expected = trimSeparators(expected)
+	assert.Nil(t, err)
+	assert.Equal(t, mockRepoPath, expected)
+
+	// test directory above cwd == repo path
+	mockRepoPath = strings.TrimSuffix(cwd, filepath.Base(cwd))
+	expected, err = getRepoExpectedLocation(ctx, mockRepoPath)
+	assert.Nil(t, err)
+	mockRepoPath = trimSeparators(mockRepoPath)
+	expected = trimSeparators(expected)
+	assert.Equal(t, mockRepoPath, expected)
+
+	// test cwd completely different from repo path
+	mockRepoPath = filepath.Join("github.com", "pulumi", "random-provider")
+	expected, err = getRepoExpectedLocation(ctx, mockRepoPath)
+	expected = trimSeparators(expected)
+	assert.Nil(t, err)
+	assert.Equal(t, filepath.Join(ctx.GoPath, "src", mockRepoPath), expected)
```

Comment:
This pattern is perfect for table based testing. That would look something like this:
```suggestion
	tests := []struct{ repoPath, expected string }{
		{ cwd, trimSeparators(expected) },
		{ strings.TrimSuffix(cwd, filepath.Base(cwd)), getRepoExpectedLocation(ctx, mockRepoPath) },
		{ filepath.Join("github.com", "pulumi", "random-provider"), trimSeparators(expected) },
	}
	
	for _, tt := range tests {
		tt := tt
		t.Run(fmt.Sprintf("(%s,%s)",tt.repoPath,tt.expected), func(t *testing.T) {
			mockRepoPath = filepath.Join("github.com", "pulumi", "random-provider")
			expected, err = getRepoExpectedLocation(ctx, mockRepoPath)
			expected = trimSeparators(expected)
			assert.Nil(t, err)
			assert.Equal(t, tt.expected, expected)
		})
	// test cwd completely different from repo path

```

Reply:
Thanks for the suggestion, applied